packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -37,6 +41,71 @@ import { buildQuery } from '../../queryBuilder';
 import { compileMetricQuery } from '../../queryCompiler';
 import { ProjectAdapter } from '../../types';
 
+const mergeMissingDbtConfigSecrets = (
```

Comment:
> wouldn't this make sense at the model level?

Yup, that makes sense. Changing it now :) 

> I was expecting this PR to change the API + service to simply pass around a Partial<APIObject> allowing any number of parameters to be changed on a PATCH request. Then the model would be responsible for merging.

That approach is more complicated because we would need to manage how to "unset" properties using `null`. I don't see a reason to implement that since we don't get any benefits from it right now. 🤔